src/System.Security.Cryptography.Xml/tests/Samples/EncryptingDecryptingSymmetricKeyWrap.cs

```diff
@@ -0,0 +1,113 @@
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml;
+using Xunit;
+
+namespace System.Security.Cryptography.Xml.Tests
+{
+    public class EncryptingDecryptingSymmetricKeyWrap
+    {
+        private static XmlDocument LoadXmlFromString(string xml)
+        {
+            var doc = new XmlDocument();
+            doc.PreserveWhitespace = true;
+            doc.LoadXml(xml);
+            return doc;
+        }
+
+        private static void Encrypt(XmlDocument doc, string elementName, string encryptionElementID, SymmetricAlgorithm key, string keyName, Func<SymmetricAlgorithm> innerKeyCtor)
+        {
+            var elementToEncrypt = (XmlElement)doc.GetElementsByTagName(elementName)[0];
+
+            using (SymmetricAlgorithm innerKey = innerKeyCtor())
+            {
+                // Encrypt the key with another key
+                var encryptedKey = new EncryptedKey()
+                {
+                    CipherData = new CipherData(EncryptedXml.EncryptKey(innerKey.Key, key)),
+                    EncryptionMethod = new EncryptionMethod(TestHelpers.GetEncryptionMethodName(key, keyWrap: true))
+                };
+
+                // Specify which EncryptedData
+                // uses this key. An XML document can have
+                // multiple EncryptedData elements that use
+                // different keys.
+                encryptedKey.AddReference(new DataReference()
+                {
+                    Uri = "#" + encryptionElementID
+                });
+
+                var encryptedData = new EncryptedData()
+                {
+                    Type = EncryptedXml.XmlEncElementUrl,
+                    Id = encryptionElementID,
+
+                    // Create an EncryptionMethod element so that the
+                    // receiver knows which algorithm to use for decryption.
+                    EncryptionMethod = new EncryptionMethod(TestHelpers.GetEncryptionMethodName(innerKey))
+                };
+
+                encryptedData.KeyInfo.AddClause(new KeyInfoEncryptedKey(encryptedKey));
+                encryptedKey.KeyInfo.AddClause(new KeyInfoName()
+                {
+                    Value = keyName
+                });
+
+                var encryptedXml = new EncryptedXml();
+                encryptedData.CipherData.CipherValue = encryptedXml.EncryptData(elementToEncrypt, innerKey, false);
+
+                EncryptedXml.ReplaceElement(elementToEncrypt, encryptedData, false);
+            }
+        }
+
+        public static void Decrypt(XmlDocument doc, SymmetricAlgorithm key, string keyName)
+        {
+            var encrypted = new EncryptedXml(doc);
+            encrypted.AddKeyNameMapping(keyName, key);
+            encrypted.DecryptDocument();
+        }
+
+        public static IEnumerable<object[]> GetSymmetricAlgorithmsPairs()
+        {
+            foreach (var first in TestHelpers.GetSymmetricAlgorithms(skipDes: true))
```

Comment:
Please add a comment in the code to that effect, so that when someone comes across it later they understand it to be a spec reason (this isn't defined) vs a workaround for a test/product reason.